libraries/botbuilder-lg/src/templates.ts

```diff
@@ -119,6 +112,37 @@ export class LGFile {
         this.options = options || [];
     }
 
+    /**
+     * Returns a new iterator for the template collection.
+     */
+    public [Symbol.iterator](): Iterator<Template> {
+        let index = 0;
+        return {
+            next: (): IteratorResult<Template> => {
+                if (index < this.items.length) {
+                    return { done: false, value: this.items[index++] };
+                } else {
+                    return { done: true, value: undefined };
+                }
+            }
+        }
+    }
+
+    /**
+     * Returns a reference to the internal list of collection templates.
+     */
+    public toArray(): Template[] {
+        return this.items;
+    }
+
+    /**
+     * Appends 1 or more templates to the collection.
+     * @param args List of templates to add.
+     */
+    public push(...args: Template[]): void {
+        args.forEach(t => this.items.push(t));
+    }
+
```

Comment:
Do you need a .length, or any other standard methods?

Reply:
No... It's not a full array so I don't wont to break into jail by adding length